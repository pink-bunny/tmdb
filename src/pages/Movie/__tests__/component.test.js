import React from 'react';
import { shallow } from 'enzyme';

import MovieComponent from '../component';

describe('MovieComponent component', () => {
  const requiredProps = {
    loading: false
  };
  const wrapper = shallow(<MovieComponent {...requiredProps} />); /* eslint-disable-line */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapsot when component is loading', () => {
    wrapper.setProps({ loading: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapsot when component has error', () => {
    wrapper.setProps({
      ...requiredProps,
      error: 'Mock error'
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapsot when component has details', () => {
    wrapper.setProps({
      ...requiredProps,
      error: '',
      details: {
        id: 1,
        title: 'Mock title',
        original_language: 'Mock language', /* eslint-disable-line camelcase */
        budget: 1000000,
        revenue: 50000,
        genres: [{ id: 1, name: 'Mock genre' }],
        watchlist: false,
        favorite: true,
        backdrops: [{}],
        cast: [{}],
        crew: [{}]
      }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
