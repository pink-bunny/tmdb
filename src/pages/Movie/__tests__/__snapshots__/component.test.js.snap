// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieComponent component matches snapshot with required props 1`] = `""`;

exports[`MovieComponent component matches snapsot when component has details 1`] = `
<Adapter>
  <Adapter>
    <Carousel
      backdrops={
        Array [
          Object {},
        ]
      }
    />
    <div
      className="top-margin"
    >
      <TopInfo
        details={
          Object {
            "backdrops": Array [
              Object {},
            ],
            "budget": 1000000,
            "cast": Array [
              Object {},
            ],
            "crew": Array [
              Object {},
            ],
            "favorite": true,
            "genres": Array [
              Object {
                "id": 1,
                "name": "Mock genre",
              },
            ],
            "id": 1,
            "original_language": "Mock language",
            "revenue": 50000,
            "title": "Mock title",
            "watchlist": false,
          }
        }
      />
      <Cast
        cast={
          Array [
            Object {},
          ]
        }
      />
      <Crew
        crew={
          Array [
            Object {},
          ]
        }
      />
    </div>
  </Adapter>
</Adapter>
`;

exports[`MovieComponent component matches snapsot when component has error 1`] = `
<Adapter>
  <Adapter>
    <Row
      className="top-margin"
      gutter={0}
      justify="center"
      type="flex"
    >
      <Col>
        <Paragraph
          type="danger"
        >
          Mock error
        </Paragraph>
      </Col>
    </Row>
  </Adapter>
</Adapter>
`;

exports[`MovieComponent component matches snapsot when component is loading 1`] = `
<Adapter>
  <Adapter>
    <Row
      className="top-margin"
      gutter={0}
      justify="center"
      type="flex"
    >
      <Col>
        <Spin
          size="default"
          spinning={true}
          wrapperClassName=""
        />
      </Col>
    </Row>
  </Adapter>
</Adapter>
`;
